conda/recipes/cudf/meta.yaml

```diff
@@ -83,7 +84,7 @@ requirements:
     - pandas >=2.0,<2.2.2dev0
     - cupy >=12.0.0
     - numba >=0.57
-    - numpy >=1.21
+    - {{ pin_compatible('numpy', max_pin='x') }}
```

Comment:
Switch to using `pin_compatible` with NumPy to ensure the minimum version allowed at runtime matches what we built with and any NumPy version (pre-2.0.0) is allowed